File: OEM/Domain/ComplementaryTasks/ResponsibleTeam.cs


namespace OEMAPI.Domain.ComplementaryTasks
{
    public class ResponsibleTeam : ValueObject
    {
        public string Value { get; }

        public ResponsibleTeam(string value)
        {
            if (string.IsNullOrWhiteSpace(value))
                throw new BusinessRuleValidationException("Responsible team cannot be empty.");

            Value = value;
        }

        protected override IEnumerable<object> GetEqualityComponents()
        {
            yield return Value;
        }
    }
}

File: OEM/src/Domain/ComplementaryTasks/ResponsibleTeam.ts
import { ValueObject } from "../../core/domain/ValueObject";
import { Result } from "../../core/logic/Result";
import { Guard } from "../../core/logic/Guard";

interface ResponsibleTeamProps {
  value: string;
}

export class ResponsibleTeam extends ValueObject<ResponsibleTeamProps> {

  get value(): string {
    return this.props.value;
  }

  private constructor(props: ResponsibleTeamProps) {
    super(props);
  }

  public static create(team: string): Result<ResponsibleTeam> {
    const guardResult = Guard.againstNullOrUndefined(team, "responsibleTeam");

    if (!guardResult.succeeded) {
      return Result.fail<ResponsibleTeam>(guardResult.message);
    }

    return Result.ok<ResponsibleTeam>(
      new ResponsibleTeam({ value: team })
    );
  }
}
